### ISSUE TITLE
Detect and update jobs in not-finished states

### ISSUE THREAD

**Issue by csomh** [packit/packit-service#1005]
For Copr builds we already have a baby-sit Celery task to periodically check the status of the builds and update the states in the database accordingly. But even so, we some builds (~0.8%) which show up in the "pending" state in the DB. For test runs, this number (jobs in some not-finished state) is even higher.

There should be a universal way to bring consitency to the database, and to also make sure, that the corresponding GitHub flags—which we use to communicate these states to our users—are in sync.

The naïve way to achive this would be to have some Celery task(s) periodically checking the status of all jobs which appear to be in a not-finished state in the DB, and update their status in the DB and GitHub in case they have changed.

Any other ideas? :slightly_smiling_face:

**Comment by TomasTomecek** [comment]
Agreed! Running such job e.g. hourly would be awesome to make sure that DB data is accurate and consistent so then we can make PR/commit statuses up to date as well.

**Comment by TomasTomecek** [comment]
Since the agreed implementation is to do it with openshift cronjobs, there should probably be a new entry in here https://github.com/packit/deployment/tree/main/cron-jobs

**Comment by lbarcziova** [comment]
Iirc we also mentioned it could be done via Celery job and we should discuss it at the arch meeting or was there already an agreement on an Openshift job?

**Comment by csomh** [comment]
Yeah, why didn't we take notes? :thinking: :smile: 

I also remember this discussion, and what I remember is that we said that in the end it might be better to do as a Celery job, to be similar to the already existing babysit one, and avoid having to set up a whole now cron-job infrastructure to run it.

**Comment by TomasTomecek** [comment]
Hmmm, the acceptance criteria of the card state to do it via k8s cron jobs so I assumed that's what agreed upon.

Let's revisit then.
